integration/metrics_test.go

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) 2021 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+
+//go:build integration
+
+package integration
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/eclipse-kanto/kanto/integration/util"
+	"github.com/eclipse-kanto/system-metrics/internal/metrics"
+
+	"github.com/eclipse/ditto-clients-golang/protocol"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	metricsFeatureID = "Metrics"
+	dataAction       = "data"
+)
+
+type systemMetricsSuite struct {
+	suite.Suite
+
+	util.SuiteInitializer
+
+	metricsThingURL   string
+	metricsFeatureURL string
+
+	pathData  string
+	topicData string
+}
+
+func (suite *systemMetricsSuite) SetupSuite() {
+	suite.SuiteInitializer.Setup(suite.T())
+
+	suite.metricsThingURL = util.GetThingURL(suite.Cfg.DigitalTwinAPIAddress, suite.ThingCfg.DeviceID)
+	suite.metricsFeatureURL = util.GetFeatureURL(suite.metricsThingURL, metricsFeatureID)
+
+	suite.pathData = util.GetFeatureOutboxMessagePath(metricsFeatureID, dataAction)
+	suite.topicData = util.GetLiveMessageTopic(suite.ThingCfg.DeviceID, protocol.TopicAction(dataAction))
+}
+
+func (suite *systemMetricsSuite) TearDownSuite() {
+	suite.SuiteInitializer.TearDown()
+}
+
+func TestSystemMetricsSuite(t *testing.T) {
+	suite.Run(t, new(systemMetricsSuite))
+}
+
+func (suite *systemMetricsSuite) testMetrics(params map[string]interface{}, originators ...string) error {
+	ws, err := util.NewDigitalTwinWSConnection(suite.Cfg)
+	require.NoError(suite.T(), err, "failed to create websocket connection")
+	defer ws.Close()
+
+	err = util.SubscribeForWSMessages(suite.Cfg, ws, "START-SEND-MESSAGES", "")
+	require.NoError(suite.T(), err, "unable to listen for events by using a websocket connection")
+
+	defer func() {
+		_, err := util.ExecuteOperation(suite.Cfg, suite.metricsFeatureURL, "request", map[string]interface{}{
+			"frequency": "0s",
+		})
+		assert.NoError(suite.T(), err, "error while stopping system metrics")
+	}()
+
+	_, err = util.ExecuteOperation(suite.Cfg, suite.metricsFeatureURL, "request", params)
```

Comment:
Current time can be collected before the request, then the metrics timestamp can be validated also.